pkg/referrerstore/oras/oras.go

```diff
@@ -135,6 +136,9 @@ func (store *orasStore) ListReferrers(ctx context.Context, subjectReference comm
 	} else {
 		resolvedSubjectDesc, err = store.GetSubjectDescriptor(ctx, subjectReference)
 		if err != nil {
+			if strings.Contains(err.Error(), "401") {
```

Comment:
Thinking more about it, its a probably safer to have logic for both  if( good) addToCache , if(bad) remove from cache. A sample scenario where ,if we are looking for 401 to remove, if somehow the request resulted gotten 500 , the bad entry will still remain in the cache. But if we only added to cache after seeing the request successful, the bad entry would never been added to the cache.

Reply:
I added `addAuthCache` and `evictAuthCache` methods which will add to the cache if all repository client operations are successful and evict if a 400, 401, or 403 error code is returned.

Reply:
ah, thanks! Looking at the implementation at https://github.com/oras-project/oras-go/blob/49bd44ce628cfedbe10ecfca76727044f0ffdaab/registry/remote/internal/errutil/errors.go#L72, return fmt.Errorf("%s %q: unexpected status code %d: %s", resp.Request.Method, resp.Request.URL, resp.StatusCode, errmsg), what is the value of resp.Request.URL. if the repostiory name contains 400 , would be catch that? Maybe it doesn't hurt to always evict on error ?

Reply:
not sure what the url contains. I'll look into that. I saw that the error message always returns 'status <status code>' in the string so now I search for that substring instead of just the error code. still not the best though...

Reply:
Discussed offline: Decided to remove the error code conditional eviction and add it as a future work item. Currently there doesn't seem to be a clean way to do this.